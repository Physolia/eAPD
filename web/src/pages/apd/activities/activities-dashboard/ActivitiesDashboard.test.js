import { shallow } from 'enzyme';
import React from 'react';

import { addActivity } from '../../../../actions/editActivity';

import {
  plain as Activities,
  mapStateToProps,
  mapDispatchToProps
} from './ActivitiesDashboard';

const initialProps = {
  addActivity: () => {},
  activities: [
    { key: 'key1', first: 'activity' },
    { key: 'key2', second: 'activity' },
    { key: 'key3', third: 'activity' }
  ]
};

jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ apdId: '0123456789abcdef01234560' })
}));

const setup = props => shallow(<Activities {...initialProps} {...props} />);

describe('the Activities component', () => {
  it('renders correctly', () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });

  it('calls addActivity when the button component is clicked', () => {
    const mockAddActivity = jest.fn();
    const component = setup({ addActivity: mockAddActivity });
    expect(mockAddActivity).not.toHaveBeenCalled();
    component.find('Button').simulate('click');
    expect(mockAddActivity).toHaveBeenCalled();
  });

  it('maps state to props', () => {
    const state = {
      apd: {
        data: {
          activities: [{ key: 'key1' }, { key: 'key2' }, { key: 'key3' }]
        }
      }
    };

    expect(mapStateToProps(state)).toEqual({
      activities: [{ key: 'key1' }, { key: 'key2' }, { key: 'key3' }]
    });
  });

  it('maps dispatch to props', () => {
    expect(mapDispatchToProps).toEqual({
      addActivity
    });
  });
});
