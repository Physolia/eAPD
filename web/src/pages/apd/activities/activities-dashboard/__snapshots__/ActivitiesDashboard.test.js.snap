// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Activities component renders correctly 1`] = `
<Fragment>
  <Connect(ConnectedWaypoint) />
   
  <Connect(AlertMissingFFY) />
  <Section
    id="activities"
    resource="activities"
  >
    <hr
      className="custom-hr"
    />
    <Connect(EntryDetails)
      activityIndex={0}
      apdId="0123456789abcdef01234560"
      key="key1"
    />
    <Connect(EntryDetails)
      activityIndex={1}
      apdId="0123456789abcdef01234560"
      key="key2"
    />
    <Connect(EntryDetails)
      activityIndex={2}
      apdId="0123456789abcdef01234560"
      key="key3"
    />
    <Button
      className="ds-u-margin-top--4"
      component="button"
      onClick={[Function]}
      type="button"
    >
      Add Activity
    </Button>
  </Section>
</Fragment>
`;
